.circleci/config.yml

```diff
@@ -53,6 +53,22 @@ restore-flow-typed-cache: &restore-flow-typed-cache
   restore_cache:
     key: *flow-typed-cache-key
 
+save-assets-cache: &save-assets-cache
+  save_cache:
+    key: v1-asset-cache-{{ .Branch }}-{{ .BuildNum }}
+    paths:
+      - public/assets
+      - public/packs-test
+      - tmp/cache/assets
+      - tmp/cache/webpacker
+
+restore-assets-cache: &restore-assets-cache
+  restore_cache:
+    keys:
+      - v1-asset-cache-{{ .Branch }}-{{ .BuildNum }}
+      # fallback to using the latest cache if no exact match is found
+      - v1-asset-cache-
```

Comment:
Reusing cache from other builds and pull requests. My understanding is that [cache is per repo](https://circleci.com/docs/2.0/caching/#caching-and-open-source), so cache generated in upstream repo will not be used for PRs submitted from forked repos as well forked repos will not share caches with each other. 

If sharing caches between PRs in same repo is undesirable, then we can remove this line and share only between same git revision.

Reply:
Yes it is cache for repo

But instead you should not have fallbacks, can lead to some unexpected errors. I prefer not caching than caching badly

Reply:
I removed catch-all key and left the branch catch latest key so we restore for the same branch at least. Is this ok with you? 

Even if assets changed since cache was created (e.g. a new commit was added to branch) things should still work because `assets:precompile` is still called. Let me know if you want me to remove also branch fallback.